pkg/storagecapabilities/storagecapabilities.go

```diff
@@ -278,6 +282,42 @@ var storageClassToProvisionerKeyMapper = map[string]func(sc *storagev1.StorageCl
 			return "UNKNOWN"
 		}
 	},
+	"csi-unity.dellemc.com": func(sc *storagev1.StorageClass) string {
+		// https://github.com/dell/csi-unity/blob/1f42af327f4130df65c5532f6029559e4ab579b5/samples/storageclass
+		switch strings.ToLower(sc.Parameters["protocol"]) {
+		case "nfs":
+			return "csi-unity.dellemc.com/nfs"
+		default:
+			return "csi-unity.dellemc.com"
+		}
+	},
+	"csi-powerstore.dellemc.com": func(sc *storagev1.StorageClass) string {
+		// https://github.com/dell/csi-powerstore/blob/76e2cb671bd3cb28aa860e9057649d1d911e1deb/samples/storageclass
+		switch strings.ToLower(sc.Parameters["csi.storage.k8s.io/fstype"]) {
+		case "nfs":
+			return "csi-powerstore.dellemc.com/nfs"
+		default:
+			return "csi-powerstore.dellemc.com"
+		}
+	},
+	"csi-vxflexos.dellemc.com": func(sc *storagev1.StorageClass) string {
+		// https://github.com/dell/csi-powerflex/tree/main/samples/storageclass
+		switch strings.ToLower(sc.Parameters["csi.storage.k8s.io/fstype"]) {
+		case "nfs":
+			return "csi-vxflexos.dellemc.com/nfs"
+		default:
+			return "csi-vxflexos.dellemc.com"
+		}
+	},
+	"csi-powermax.dellemc.com": func(sc *storagev1.StorageClass) string {
```

Comment:
If it's ok now I'll try to add the last three to a function, because they have the same string key (`csi.storage.k8s.io/fstype`)